attribute/set.go

```diff
@@ -53,16 +58,44 @@ type (
 	Sortable []KeyValue
 )
 
-var (
-	// keyValueType is used in computeDistinctReflect.
-	keyValueType = reflect.TypeOf(KeyValue{})
+var slicePool = sync.Pool{New: func() any { return new([]KeyValue) }}
 
-	// emptySet is returned for empty attribute sets.
-	emptySet = &Set{
-		equivalent: Distinct{
-			iface: [0]KeyValue{},
-		},
+func getSlice(length, capacity int) *[]KeyValue {
+	v := slicePool.Get().(*[]KeyValue)
+	if cap(*v) < capacity {
+		*v = make([]KeyValue, length, capacity)
 	}
+	*v = (*v)[:length]
+	return v
+}
+
+// sets is a registry of all Set data.
+var sets = newSetRegistry(-1)
+
+func newSet(data *[]KeyValue) Set {
+	id := getID()
```

Comment:
I am not quite sure but is it necessary to allocate (or get from a pool) a pointer to an id? The id is already stored in the `setData.key`. Can Set.id point to `setData.key` instead?